### PR TITLE
Update server node dependencies

### PR DESCRIPTION
**Pull request by BryanQuigley**
Was looking for updates to help with the memory usage. Didn't succeed, but these updates do fix some security notices.

@NestJS
@Type
@turf
@types (except node which set to 16)

Green updates - @nestjs-modules/mailer bcrypt eslint-plugin-import pg

Yellow - @typescript-eslint/eslint-plugin @typescript-eslint/parser aws-sdk
eslint eslint-config-prettier eslint-plugin-functional lru-cache rollbar
rxjs ts-node tsconfig-paths

Red
Passport, geojson2shp, eslint-plugin-jsdoc, class-transformer, jest

Dropped
@godaddy/terminus
ts-loader

### CODE REVIEW

File: src/server/package.json [hunk 1]
```diff
@@ -26,88 +26,86 @@
   },
   "dependencies": {
     "@drdgvhbh/postgres-error-codes": "0.0.6",
-    "@godaddy/terminus": "4.3.1",
-    "@nestjs-modules/mailer": "1.6.0",
-    "@nestjs/common": "8.2.4",
-    "@nestjs/core": "8.2.4",
+    "@nestjs-modules/mailer": "1.6.1",
```
Comment (maurizi) [inline src/server/package.json:29]: This update broke sending emails.
I had to make the following change to get them working again with this version:
```diff
diff --git a/src/server/src/auth/services/auth.service.ts b/src/server/src/auth/services/auth.service.ts
index 3fe19b42..585a90fc 100644
--- a/src/server/src/auth/services/auth.service.ts
+++ b/src/server/src/auth/services/auth.service.ts
@@ -104,7 +104,7 @@ export class AuthService {
           ? `${process.env.CLIENT_URL}/activate/${emailToken}/${org}`
           : `${process.env.CLIENT_URL}/activate/${emailToken}`
       },
-      template: "./verify"
+      template: "verify"
     });
 
     this.logSentEmail(info);
@@ -123,7 +123,7 @@ export class AuthService {
         user,
         url: `${process.env.CLIENT_URL}/password-reset/${emailToken}`
       },
-      template: "./passwordreset"
+      template: "passwordreset"
     });
 
     this.logSentEmail(info);
```

**Review by maurizi — CHANGES_REQUESTED** [review]
Tested and working other than the the two regressions I found & left comments about.

As per my other comment, I'd prefer to either drop the `eslint` updates here or also include updates to `eslint` elsewhere in the project.

File: src/server/package.json [hunk 2]
```diff
@@ -26,88 +26,86 @@
...
+    "@types/supertest": "2.0.11",
+    "@types/topojson-client": "3.1.0",
+    "@types/validator": "13.7.1",
+    "@typescript-eslint/eslint-plugin": "5.15.0",
+    "@typescript-eslint/parser": "5.15.0",
+    "eslint": "8.11.0",
+    "eslint-config-prettier": "8.5.0",
+    "eslint-plugin-functional": "4.2.0",
+    "eslint-plugin-import": "2.25.4",
+    "eslint-plugin-jsdoc": "38.0.3",
```
[the remark this answers is not in the document]
Comment (BryanQuigley) [reply src/server/package.json]: updated them all

**Comment by BryanQuigley** [comment]
made eslint changes and condensed, but haven't tested locally yet.

**Review by maurizi — APPROVED** [review]
👍

**Pull request merged**